File: movie-infrastructures/storage/src/main/java/com/movie/storage/movie/repository/TheaterRepositoryImpl.java
package com.movie.storage.movie.repository;

import com.movie.domain.movie.TheaterRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class TheaterRepositoryImpl implements TheaterRepository {

    private final TheaterJpaRepository theaterJpaRepository;

}

File: movie/infra/src/main/kotlin/com/example/movie/persistence/theater/repository/TheaterRepositoryImpl.kt
package com.example.movie.persistence.theater.repository

import com.example.movie.domain.theater.model.Theater
import com.example.movie.domain.theater.repository.TheaterRepository
import org.springframework.stereotype.Repository

@Repository
class TheaterRepositoryImpl(
    private val theaterJpaRepository: TheaterJpaRepository
) : TheaterRepository {
    override fun findById(id: Long): Theater? {
        return theaterJpaRepository.findById(id).orElse(null)?.toDomain()
    }

    override fun findAll(): List<Theater> {
        return theaterJpaRepository.findAll().map { it.toDomain() }
    }
}